src/linear_algebra/finite_dimensional.lean

```diff
@@ -828,6 +828,78 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma eq_zero_of_findim_eq_zero [finite_dimensional K V]
+  (hV : findim K V = 0) (x : V) : x = 0 :=
+begin
+  obtain ⟨B, hB⟩ := exists_is_basis_finset K V,
+  have := (findim_eq_card_finset_basis hB).symm,
+  rw [hV, finset.card_eq_zero, finset.eq_empty_iff_forall_not_mem] at this,
+  rw ← is_basis.equiv_fun_total hB x,
+  convert finset.sum_empty,
+  rw finset.univ_eq_empty,
+  rintro ⟨b, hB⟩,
+  exact this b hB,
+end
+
+lemma subsingleton.is_basis (h : subsingleton V) : is_basis K (λ x : fin 0, (0 : V)) :=
+begin
+  split,
+  { rw linear_independent_iff,
+    intros l hl,
+    ext x,
+    exact fin.elim0 x },
+  { rw eq_top_iff,
+    intros x _,
+    rw [subsingleton.elim x 0],
+    exact zero_mem _ }
+end
```

Comment:
This lemma is `is_basis_empty V (sorry : \not nonempty (fin 0))` once #6281 is merged

Reply:
I will change to a simpler proof using `is_basis_empty` now and will simplify it even further once your pr is merged